.github/workflows/build-wheels-linux.yml

```diff
@@ -0,0 +1,57 @@
+# From https://github.com/pytorch/test-infra/wiki/Using-Nova-Reusable-Build-Workflows
+name: Build Linux Wheels
+
+on:
+  pull_request:
+    paths:
+      - build/packaging/**
+      - .github/workflows/build-wheels-linux.yml
+  push:
+    branches:
+      - nightly
+      - main
+      - release/*
+    tags:
+      # NOTE: Binary build pipelines should only get triggered on release candidate builds
+      # Release candidate tags look like: v1.11.0-rc1
+      - v[0-9]+.[0-9]+.[0-9]+-rc[0-9]+
+  workflow_dispatch:
+
+jobs:
+  generate-matrix:
+    uses: pytorch/test-infra/.github/workflows/generate_binary_build_matrix.yml@main
+    with:
+      package-type: wheel
+      os: linux
+      test-infra-repository: pytorch/test-infra
+      test-infra-ref: main
+      with-cuda: disabled
+      with-rocm: disabled
+
+  build:
+    needs: generate-matrix
+    permissions:
+      id-token: write
+      contents: read
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - repository: pytorch/executorch
+            pre-script: build/packaging/pre_build_script.sh
+            post-script: build/packaging/post_build_script.sh
+            smoke-test-script: build/packaging/smoke_test.py
```

Comment:
I know @dbort is working on the providing the pip package so that users could grab it via `pip install executorch` as well as the runtime lib libexecutorch linking potentially with libkernels and libdelegate, etc, which seems to be wrapped in these pre/post build scripts.

Reply:
It would be great to see that part integrating with these CI scripts.  Here is what the general build process looks like for PyTorch and other domains libraries.

1. Run pre-script https://github.com/pytorch/executorch/actions/runs/8243162791/job/22543286939#step:12:111
2. Build the wheel file with `python setup.py bdist_wheel` https://github.com/pytorch/executorch/actions/runs/8243162791/job/22543286939#step:14:59
3. Run the post-script https://github.com/pytorch/executorch/actions/runs/8243162791/job/22543286939#step:15:110
4. Install the wheel from step 2
5. Run the smoke test script to verify that the wheel looks right.

Although the current ExecuTorch wheel doesn't have everything it needs yet, they could be added later.  Once this land, I can add a trigger on https://github.com/pytorch/test-infra/blob/main/.github/workflows/trigger_nightly.yml to see it works end to end.